client/tink_keyset_helper.go

```diff
@@ -119,3 +130,97 @@ func readTinkKeysetFromBytes(data []byte) (*tinkpb.Keyset, error) {
 	}
 	return tinkKeyset, nil
 }
+
+// getTinkKeysetHandleFromKnoxVersionList returns a tink keyset handle that has all tink keys in the
+// received knox version list and a map from tink key IDs to knox version IDs. Since each version is
+// a tink keyset that contains a single key, this func will enumerate all versions in the given version
+// list and do the appending.
+func getTinkKeysetHandleFromKnoxVersionList(
+	knoxVersionList knox.KeyVersionList,
+) (*keyset.Handle, map[uint32]uint64, error) {
+	tinkKeyset := new(tinkpb.Keyset)
+	tinkKeyIDToKnoxVersionID := make(map[uint32]uint64)
+	for _, v := range knoxVersionList {
+		// the data of each version is a tink keyset that contains a single tink key
+		keysetContainsASingleKey, err := readTinkKeysetFromBytes(v.Data)
+		if err != nil {
+			return nil, nil, err
+		}
+		singleKey := keysetContainsASingleKey.Key[0]
+		if v.Status == knox.Primary {
+			tinkKeyset.PrimaryKeyId = singleKey.KeyId
+		}
+		tinkKeyset.Key = append(tinkKeyset.Key, singleKey)
+		tinkKeyIDToKnoxVersionID[singleKey.KeyId] = v.ID
+	}
+	// Convert tink keyset to tink keyset handle. tink doesn't allow transferring cleartext keyset to keyset
+	// handle driectly. Hence, keyset is converted to bytes, then read by package insecurecleartextkeyset
+	bytesBuffer := new(bytes.Buffer)
+	writer := keyset.NewBinaryWriter(bytesBuffer)
+	writer.Write(tinkKeyset)
+	reader := keyset.NewBinaryReader(bytesBuffer)
+	keysetHandle, err := insecurecleartextkeyset.Read(reader)
```

Comment:
@kalpeshdharwadkar  is it alright to be using this tink package? I'm not familiar with the details--I'm assuming so given Knox's wrapping around this.

Reply:
Tink project wants to enforce encrypted DEK on disk and decrypting them in the memory using KEK stored in KMS.  Hence, they have prefixed the functions that facilitates reading/writing cleartext DEK from/to disk as insecure. We need to use these functions as we do not leverage KMS directly on the client but read/write DEK from/to knox. We would also need this function while parsing DEK from Knox File cache.